asynq/tests/test_tools.py

```diff
@@ -384,3 +384,28 @@ def async_trigger():
     with AssertRaises(AssertionError):
         hook3.safe_trigger()
     assert_eq(['handler2()'], calls)
+
+
+class DeduplicateClassWrapper:
+    @deduplicate()
+    @asynq()
+    def return_three(self):
```

Comment:
Is it worth having a test where we stack `@deduplicate` and `@acached_per_instance`

Reply:
i'm pretty sure the problem you had actually had nothing to do with acached_per_instance -- it repros even without those decorators